Firestore/Source/Remote/FSTDatastore.m

```diff
@@ -436,10 +438,56 @@ + (void)prepareHeadersForRPC:(GRPCCall *)rpc
 
 @end
 
+#pragma mark - GRXFilter
```

Comment:
This file is getting ridiculously huge. I think we should split the streams out into separate files. WDYT?

I'm not asking for that as a part of this change, but what do you think of starting on that by putting this filter in its own files?

Reply:
I totally agree. I'd suggest doing this in one go in a follow-up PR though.